test/stream/bridge/keboola/keboola_test.go

```diff
@@ -82,162 +65,249 @@ func TestKeboolaBridgeWorkflow(t *testing.T) {
 	ts := setup(t, ctx, configFn)
 	defer ts.teardown(t, ctx)
 
-	ts.testSlicesUpload(
-		t,
-		ctx,
-		sliceUpload{
-			0,
-			20,
-			[]model.FileState{
-				model.FileWriting,
-			},
-			[]model.SliceState{
-				model.SliceWriting,
-				model.SliceWriting,
+	// Check initial state
+	ts.checkState(t, ctx, []file{
+		{
+			state: model.FileWriting,
+			volumes: []volume{
+				{
+					slices: []model.SliceState{
+						model.SliceWriting,
+					},
+				},
```

Comment:
E2E test streams/Keboola was unstable and there were several issues in it - like we didn't finish properly under time stress, ... there was a lot of complicated math about which slice index is checked and so on.

I suggest to ook at the entire test, not the diff.

It should be much clearer what is happening and how the conditions are changing.